src/storage/src/source/mod.rs

```diff
@@ -789,6 +836,19 @@ where
             };
 
             let mut source_upper = timestamper.source_upper_at(upper_ts.saturating_sub(1));
```

Comment:
Is this `ReclockOperator` only used for this one `source_upper` call? if so, perhaps it can be passed in instead